packages/build/src/plugins_core/secrets_scanning/utils.ts

```diff
@@ -98,49 +122,75 @@ export function getSecretKeysToScanFor(env: Record<string, unknown>, secretKeys:
-    const val = env[key]
-    if (isValueTrivial(val)) {
-      return false
-    }
-    return isLikelySecretValue(val)
-  })
-  return nonSecretKeysToScanFor
-}
-
-const LIKELY_SECRET_MIN_LENGTH = 16
+const ENHANCED_MATCH_PREFIX_LENGTH = 4
+// Most prefixes are 4-5 chars, so requiring 12 chars after ensures a reasonable secret length
+const MIN_CHARS_AFTER_PREFIX = 12
 
 /**
- * When the enhanced secret scan is run, we check any env vars _not_ marked as secret if they are highly likely to be secret values.
- * For now, this means the value is a string of at least 16 chars and starts with one of the known prefixes.
+ * Checks a line of text for likely secrets based on known prefixes and patterns.
+ * The function works by:
+ * 1. Splitting the line into tokens using quotes, whitespace, equals signs, colons, and commas as delimiters
+ * 2. For each token, checking if it matches our secret pattern:
+ *    - Must start (^) with one of our known prefixes (e.g. aws_, github_pat_, etc)
+ *    - Must be followed by at least MIN_CHARS_AFTER_PREFIX non-whitespace characters
+ *    - Must extend to the end ($) of the token
  *
- * @param val env var value
- * @returns boolean
+ * For example, given the line: secretKey='aws_123456789012345678'
+ * 1. It's split into tokens: ['secretKey', 'aws_123456789012345678']
+ * 2. Each token is checked against the regex pattern:
+ *    - 'secretKey' doesn't match (doesn't start with a known prefix)
+ *    - 'aws_123456789012345678' matches (starts with 'aws_' and has sufficient length)
+ *
+ * @param line The line of text to check
+ * @param file The file path where this line was found
+ * @param lineNumber The line number in the file
+ * @param omitValuesFromEnhancedScan Optional array of values to exclude from matching
+ * @returns Array of matches found in the line
  */
-function isLikelySecretValue(val): boolean {
-  if (typeof val !== 'string') {
-    return false
-  }
-  if (val.length < LIKELY_SECRET_MIN_LENGTH) {
-    return false
-  }
-  if (LIKELY_SECRET_PREFIXES.some((prefix) => val.toLowerCase().startsWith(prefix))) {
-    return true
+export function findLikelySecrets({
+  line,
+  file,
+  lineNumber,
+  omitValuesFromEnhancedScan = [],
+}: {
+  line: string
+  file: string
+  lineNumber: number
+  omitValuesFromEnhancedScan?: unknown[]
+}): MatchResult[] {
+  if (!line) return []
+
+  // Escape special regex characters (like $, *, +, etc) in prefixes so they're treated as literal characters
+  const prefixPattern = LIKELY_SECRET_PREFIXES.map((p) => p.replace(/[$*+?.()|[\]{}]/g, '\\$&')).join('|')
+  // Build regex pattern:
+  // ^ - match start of token
+  // (?:${prefixPattern}) - non-capturing group containing our prefixes (e.g. aws_|github_pat_|etc)
+  // [^\\s]{${MIN_CHARS_AFTER_PREFIX},} - at least MIN_CHARS_AFTER_PREFIX non-whitespace chars
+  // $ - match end of token
+  // i - case insensitive flag
+  const regex = new RegExp(`^(?:${prefixPattern})[^\\s]{${MIN_CHARS_AFTER_PREFIX},}$`, 'i')
```

Comment:
This allocates those same regexes every time for each line - would it be possible to generate those regexes once and then on each line we just reuse same regex instance and just run `regex.test(line)`?

At least right now those regexes seems really static, so they might be just moved out of function scope to module scope to init them once?